import axiosInstance from "@/lib/axios";
import { getCurrentUser } from "@/hooks/auth/auth";
import { AxiosError } from "axios";

export const correccionService = {
  //Actualizar correcci√≥n
  async actualizarCorreccion(documentId: string, nuevaDescripcion: string, comentario: string) {
    const user = await getCurrentUser();

    const payload = {
      data: {
        textoGenerado: nuevaDescripcion,
        esModificada: true,
        comentarioModif: comentario,
        modificadoPor: user.email,
      },
    };

    console.log("üì¶ Payload de actualizaci√≥n de correcci√≥n:", payload);

    try {
      const response = await axiosInstance.put(`/correcciones/${documentId}`, payload);
      console.log("‚úîÔ∏è Correccion acutalizada", response.data.data);
      return response.data.data;
    } catch (err: unknown) {
      if ((err as AxiosError).isAxiosError) {
        const axiosErr = err as AxiosError;
        console.error(
          "‚ùå Error actualizando correcci√≥n:",
          axiosErr.response?.data || axiosErr.message
        );
      } else {
        console.error("‚ùå Error inesperado:", err);
      }
      throw err;
    }
  },

  async actualizarEsAceptada(documentId: string, estado: boolean) {
    console.log("Actualizando Correcci√≥n - Estado esAceptada: ", estado);

    const payload = {
      data: {
        esAceptada: estado,
      },
    };

    try {
      await axiosInstance.put(`/correcciones/${documentId}`, payload);
    } catch (err: unknown) {
      if ((err as AxiosError).isAxiosError) {
        const axiosErr = err as AxiosError;
        console.error(
          "‚ùå Error al actualizar estado de aceptaci√≥n:",
          axiosErr.response?.data || axiosErr.message
        );
      } else {
        console.error("‚ùå Error inesperado:", err);
      }
      throw err;
    }
  },
};
